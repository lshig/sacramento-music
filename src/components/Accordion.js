import React from 'react';
import options from '../data/project';

export default function Accordian() {
  return (
    <div className="project-info">
      <ul className="collapsible" data-collapsible="accordion">
        {options.map((item, key) => {
          return (
            <li key={key}>
              <div className="collapsible-header">
                <i className={item.icon} />
                <span>{item.title}</span>
              </div>
              <div className="collapsible-body">
                <pre className="pre-wrap">{item.description}</pre>
              </div>
            </li>
          );
        })}
      </ul>
    </div>
  );
}
