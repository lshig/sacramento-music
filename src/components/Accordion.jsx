import React from 'react';
import options from '../constants/ProjectData';
export default function Accordian () {
  return (
    <div className="project-info">
      <ul className="collapsible" data-collapsible="accordion">
        {options.map(item => {
          return (
            <li key={item.id + '-item'}>
              <div className="collapsible-header">
                <i className={item.icon}></i>
                <span>{item.title}</span>
              </div>
              <div className="collapsible-body">
                <pre className="pre-wrap">{item.description}</pre>
              </div>
            </li>
          )
        })}
      </ul>
    </div>
  );
}
