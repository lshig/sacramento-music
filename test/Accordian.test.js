import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Accordion from '../src/components/Accordion';

describe('Accordion', () => {
  describe('basic', () => {
    const wrapper = shallow(<Accordion />);

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
