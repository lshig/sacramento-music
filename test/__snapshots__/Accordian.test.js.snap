// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion basic renders 1`] = `
<div
  className="project-info"
>
  <ul
    className="collapsible"
    data-collapsible="accordion"
  >
    <li
      key="0"
    >
      <div
        className="collapsible-header"
      >
        <i
          className="fa fa-map-marker"
        />
        <span>
          Sacramento, CA
        </span>
      </div>
      <div
        className="collapsible-body"
      >
        <pre
          className="pre-wrap"
        >
          It's the capital of California and the Most Diverse and Integrated City in the United States. From 2009-2012, Sacramento, CA drafted into Forbes' America's Most Miserable Cities list as a product of rising home foreclosures, hiked state income taxes, low-ranking Sacramento Kings seasons, and the team's liklihood for relocating. In early 2013, #SacramentoProud circulated as a fan movement to keep the Kings in the city. By mid-2013, the city and the Kings organization closed the deal to transfer majority ownership and to construct the new Golden 1 Center. As of October 2016, the new arena carries the hopes of attracting business to the Downtown Sacramento region.

Being 916 born-n-raised, I understand this city's struggles, the loyalty of its community, and its gritty underdog spirit. It's not The Bay or LA; IT'S HOME.
        </pre>
      </div>
    </li>
    <li
      key="1"
    >
      <div
        className="collapsible-header"
      >
        <i
          className="fa fa-music"
        />
        <span>
          Music
        </span>
      </div>
      <div
        className="collapsible-body"
      >
        <pre
          className="pre-wrap"
        >
          I listen to music all day, every day. I love how sound and poetry canvases creativity, emotions, and ideas. For this project, I specifically sample and highlight songs/artists that give Sacramento, CA a shout out. It's limited to 251 songs with 339 artists, but will hopefully grow over time as I discover more and automate the process.
        </pre>
      </div>
    </li>
    <li
      key="2"
    >
      <div
        className="collapsible-header"
      >
        <i
          className="fa fa-code"
        />
        <span>
          Code
        </span>
      </div>
      <div
        className="collapsible-body"
      >
        <pre
          className="pre-wrap"
        >
          • Git + GitHub
• Atom + atom-beautify + sass-autocompile + jshint
• Node.js + npm
• Bower
• SASS
• Babel + ES2015
• Webpack
• React + jsx
• Materialize
• Font Awesome
• Highcharts + react-highcharts
        </pre>
      </div>
    </li>
    <li
      key="3"
    >
      <div
        className="collapsible-header"
      >
        <i
          className="fa fa-gift"
        />
        <span>
          Project
        </span>
      </div>
      <div
        className="collapsible-body"
      >
        <pre
          className="pre-wrap"
        >
          Before this project, I knew a handful of songs that reference Sacramento, CA:

• "California Love" by 2Pac
• "Area Codes" by Ludacris
• "This is Why I'm Hot" by MIMS
• "I Don't Fuck With You" by Big Sean ft E-40
• "New Bae" by Vic Mensa

I thought, "There has to be more than 5 songs out there!" From June 22 to July 5, 2016, I collected data and created this visual explanation of my findings. As musical artists give shout outs through lyrics, this project is my creative outlet to give a shout out and uniquely intersect Sacramento, CA, music, and code.
        </pre>
      </div>
    </li>
    <li
      key="4"
    >
      <div
        className="collapsible-header"
      >
        <i
          className="fa fa-exclamation-triangle"
        />
        <span>
          Disclaimers
        </span>
      </div>
      <div
        className="collapsible-body"
      >
        <pre
          className="pre-wrap"
        >
          No guarantee on accuracy; this project is for leisure and entertainment. It's given to the public as is.

Time: I gave myself leisure data collection effort from June 22 to July 5, 2016. The data in this project is a sample size limited by my efforts. Thus, not enough time to actually read all lyrics, check the context of the token words, and listen to the songs for accuracy. There are songs out there that probably existed prior to the time of the Internet. All of the data for this project requires that they exist on the Internet and I could find it through search engines.

Ownership: There are no gaurantees on who owns the lyrics, who performed a cover on the song, or who sampled the song.

Definition: Sacramento is a Spanish word for sacrament, potentially used in the context of Catholicism. Some artists or songs may contain "sacramento" but intended it in the Spanish sense.

Language: Only songs that include lyrics and are mostly in English were accepted for this sample size. Unfortunately some Spanish, Portuguese, French, and Hmong songs were rejected in the case that I may misinterpret the artist's lyrics.

Spelling and Slang: Lyrics are interpreted and limited by my ability to decipher the spelling/slang of the author who contributed. Whether the author chose their spelling as a matter of creativity or accidentally, it made searching and parsing time consuming. For example, Sactown vs Sac-town vs Sac Town vs Sacktown.

Nicknames: According to Wikipedia, Sacramento, CA has the following common nicknames: City of Trees, Sactown, Sac, Sacto, Sack-a-Tomatoes, Sac-a-Cement-o. However, as I continued along this project's endeavors, I found several creative variations like Jack-Ramento, Sac-A-Indo, and of course the 916 (or Nine One Six).

Literals + Symbolism: There are literal city/county references within a name, but there are clear symbols that reference Sacramento, CA. For example, Mack Road, Sacramento City College, Sacramento Kings, the Capitol, capital of California. As a result, data was categorized by city/county, street/neighborhood, sports, education, landmarks, rivers, and everything else.

Location: There's Sacramento the city, Sacramento County, Sacramento Valley region, and the Greater Sacramento metro area. Although many artists may be referring to the city, depending on the audience, many people reference Sacramento, CA even if they are positioned in a neighboring city, county, region, or metro area. Also, there are at least 31 locations around the world named Sacramento, including many in South American countries. For this project I examined the surrounding context of the lyrics and if an artist has relations or origins in California.
        </pre>
      </div>
    </li>
  </ul>
</div>
`;
